pkg/runtime/serializer/protobuf_extension.go

```diff
@@ -0,0 +1,42 @@
+// +build proto
+
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package serializer
+
+import (
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/runtime/serializer/protobuf"
+)
+
+const defaultProtobufContentType = "application/vnd.kubernetes.protobuf"
```

Comment:
I'll rename but going to leave it private for now until we know who needs
to use it.  We may end up having it in pkg/api eventually and then people
would depend on it there.

On Thu, Mar 31, 2016 at 10:01 AM, Wojciech Tyczynski <
notifications@github.com> wrote:

> In pkg/runtime/serializer/protobuf_extension.go
> <https://github.com/kubernetes/kubernetes/pull/20377#discussion_r58057861>
> :
>
> > +
> > +Unless required by applicable law or agreed to in writing, software
> > +distributed under the License is distributed on an "AS IS" BASIS,
> > +WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
> > +See the License for the specific language governing permissions and
> > +limitations under the License.
> > +*/
> > +
> > +package serializer
> > +
> > +import (
> > +	"k8s.io/kubernetes/pkg/runtime"
> > +	"k8s.io/kubernetes/pkg/runtime/serializer/protobuf"
> > +)
> > +
> > +const defaultProtobufContentType = "application/vnd.kubernetes.protobuf"
>
> OK - that makes sense.
> How about renaming it to ContentTypeProtobuf (for consistency with
> ContentTypeJSON)?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/20377/files/d7a1e247a94bf7eef7f493d96d603644e664d365#r58057861>
>